### PR TITLE
fixes broken loading of UI resources

### LINKED ISSUE #3694: java 21 java -jar zipkin.jar
## Describe the Bug
java.lang.IllegalArgumentException: Unsupported URL: jar:nested:/root/zipkin.jar/!BOOT-INF/lib/zipkin-lens-3.0.3.jar!/zipkin-lens/static/js/main.d21e7047.js (must start with 'file:', 'jar:file', 'jrt:' or 'bundle:').
And the ui load error.

## Steps to Reproduce
curl -sSL https://zipkin.io/quickstart.sh | bash -s
java -jar zipkin.jar

java -version
openjdk version "21.0.1" 2023-10-17 LTS
OpenJDK Runtime Environment Corretto-21.0.1.12.1 (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Corretto-21.0.1.12.1 (build 21.0.1+12-LTS, mixed mode, sharing)
## Expected Behaviour
ui showed

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Fixes #3694 until it can be addressed upstream by https://github.com/line/armeria/pull/5391

Until this is in, the UI is completely busted

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
added an integration test which tests this via docker, ensuring either jar or jar via docker don't have a bug that prevents loading the UI.

File: zipkin-server/src/main/java/com/linecorp/armeria/server/file/HttpFile.java [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+        if ("file".equals(url.getProtocol())) {
+            File f;
+            try {
+                f = new File(url.toURI());
+            } catch (URISyntaxException ignored) {
+                f = new File(url.getPath());
+            }
+
+            return builder(f.toPath());
+        } else if ("jar".equals(url.getProtocol()) && (url.getPath().startsWith("file:")||url.getPath().startsWith("nested:")) ||
```
Comment (codefromthecrypt) [inline zipkin-server/src/main/java/com/linecorp/armeria/server/file/HttpFile.java:203]: this is the patched line from https://github.com/line/armeria/pull/5391

**Comment by codefromthecrypt** [comment]
since zipkin is unusable, I'm going to merge and cut this. We can revert it once there's a fix upstream.

**Comment by shakuzen** [comment]
Late but LGTM

**Pull request merged**